src/components/FilterPanel.jsx

```diff
@@ -70,12 +100,36 @@ export default function FilterPanel({ onClose }) {
             expandIcon={<ExpandMoreIcon />}
             aria-controls={name + '-content'}
             id={name + '-header'}>
-            <Typography>{name}</Typography>
+            <Grid container alignItems="center" justifyContent="space-between">
+              <Typography>{name}</Typography>
+              <Badge
+                badgeContent={filter.badge}
```

Comment:
I'm noticing the badge doesn't appear or disappear until filters are applied. There's also no indicator that filters have or haven't been applied. Some changes to look into:
- Have Badge (or Chip) indicate the number of filters applied/not yet applied. E.g. It could say `3 not yet applied` vs. `2 applied` or simply `2`. 
- The badge could also be a different color if the filter value is different from the last applied value.
- The `Apply Filters` button could be enabled only when there are changes to be applied.
- We could have some text above/below the `Apply Filters` button that indicates filters haven't been applied yet, e.g. 'Your filters have not been applied yet'.
- On mobile before we allow the filter panel to close we should provide a warning confirmation dialog about leaving without applying changes (in which case the local state of the filters here should reset).

More reading: https://baymard.com/blog/how-to-design-applied-filters